dftimewolf/lib/exporters/s3_to_gcs.py

```diff
@@ -0,0 +1,155 @@
+    """
+    super(S3ToGCSCopy, self).__init__(
+        state, name=name, critical=critical)
+    self.aws_region: str = ''
+    self.dest_project_name: str = ''
+    self.dest_project: gcp_project.GoogleCloudProject = ''
+    self.dest_bucket: str = ''
+    self.filter: Any = None
+
+  # pylint: disable=arguments-differ
+  def SetUp(self,
+            aws_region: str,
+            dest_project: str,
+            dest_bucket: str,
+            s3_objects: str='',
+            object_filter: str='') -> None:
+    """Sets up a copy operation from AWS S3 to GCP GCS.
+
+    AWS objects to copy are sourced from either the state, or passed in here.
+    Args:
+      aws_region (str): The AWS region (for account.AWSAccount creation).
+      dest_project (str): The destination GCP project.
+      dest_bucket (str): The destination GCP bucket.
+      s3_objects (str): Comma separated list of objects to copy from S3. Each
+        should be of the form 's3://bucket-name/path/to/object'
+      object_filter (str): regex filter for objects to copy - Useful when the
+        files are from a previous module but not all should be transferred.
+    """
+    self.aws_region = aws_region
+    self.dest_project_name = dest_project
+    self.dest_bucket = dest_bucket
+    self.dest_project = gcp_project.GoogleCloudProject(self.dest_project_name)
+    self.filter = object_filter
+
+    if s3_objects:
+      for obj in s3_objects.split(','):
+        self.state.StoreContainer(containers.AWSS3Object(obj))
+
+  def PreProcess(self) -> None:
+    """Prep work for copying objects from S3 to GCS - Bucket creation."""
+    # Check if the destination bucket exists. If not, create it.
+    if not self.dest_bucket:
+      self.ModuleError('No destination GCP bucket specified', critical=True)
+    if self.dest_bucket not in \
+        [bucket['id'] for bucket in self.dest_project.storage.ListBuckets()]:
+      self.logger.info('Creating GCS bucket {0:s}'.format(self.dest_bucket))
+      self.dest_project.storage.CreateBucket(self.dest_bucket)
+
+    # Set the permissions on the bucket
+    self.logger.info('Applying permissions to bucket')
+    self._SetBucketServiceAccountPermissions()
+
+  def Process(self, container: containers.AWSS3Object) -> None:
+    """Creates and exports disk image to the output bucket."""
+    if self.filter:
+      if not re.match(self.filter, container.path):
+        self.logger.info('{0:s} does not match filter. Skipping.'\
+            .format(container.path))
+        return
+
+    # We must create a new client for each thread, rather than use the class
+    # member self.dest_project due to an underlying thread safety issue in
+    # httplib2: https://github.com/googleapis/google-cloud-python/issues/3501
+    client = gcp_project.GoogleCloudProject(self.dest_project_name)
+    client.storagetransfer.S3ToGCS(
+        container.path, self.aws_region + 'a', 'gs://' + self.dest_bucket)
```

Comment:
What is the `+ 'a'` ?

Reply:
Grab the first AWS Availability Zone in the region. AWS availability zones are named for the region, appended with a, b, c...